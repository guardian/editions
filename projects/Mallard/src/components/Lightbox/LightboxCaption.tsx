import React from 'react';
import { Platform, StyleSheet, View } from 'react-native';
import RenderHtml from 'react-native-render-html';
import { Direction } from '../../../../Apps/common/src';
import { ArticleTheme } from '../../components/article/article';
import { themeColors } from '../../components/article/helpers/css';
import { Arrow } from '../../components/icons/Arrow';
import { families } from '../../theme/typography';

const styles = StyleSheet.create({
	captionWrapper: {
		fontFamily: families.text.regular,
		position: 'absolute',
		zIndex: 1,
		opacity: 0.8,
		backgroundColor: themeColors(ArticleTheme.Dark).background,
		bottom: 0,
		width: '100%',
		padding: 10,
	},
	captionText: {
		color: themeColors(ArticleTheme.Dark).text,
		paddingLeft: 2,
		paddingRight: 13,
	},
	caption: {
		display: 'flex',
		flex: 1,
		flexDirection: 'row',
		paddingHorizontal: 10,
		paddingBottom: 50,
	},
});

const captionStyleSheet = (pillarColor: string) => ({
	span: {
		fontFamily: families.sans.regular,
		color: themeColors(ArticleTheme.Dark).text,
		fontSize: Platform.OS === 'android' ? 16 : 14,
	},
	b: {
		fontFamily: families.sans.bold,
		color: themeColors(ArticleTheme.Dark).text,
	},
	strong: {
		fontFamily: families.sans.bold,
		color: themeColors(ArticleTheme.Dark).text,
		paddingLeft: 12,
	},
	a: {
		color: pillarColor,
		textDecoration: 'underline',
	},
	em: {
		fontFamily: families.sans.regularItalic,
	},
});

const LightboxCaption = ({
	caption,
	pillarColor,
	displayCredit,
	credit,
}: {
	caption: string;
	pillarColor: string;
	displayCredit?: boolean;
	credit?: string;
}) => {
	const captionText = () => {
		if (displayCredit === true && credit) {
			return caption + ' ' + credit;
		} else {
			return caption;
		}
	};

	const captionStyles = captionStyleSheet(pillarColor);
	const captionSource = { html: `<span>${captionText()}</span>` };
	console.log(captionStyles);
	console.log(captionSource);

	return (
		<View style={styles.captionWrapper}>
			<View style={styles.caption}>
				{caption.length > 1 && (
					<Arrow
						fill={pillarColor}
						direction={Direction.top}
						marginTop={Platform.OS === 'android' ? 10 : 4}
					/>
				)}
				<View style={styles.captionText}>
					<RenderHtml
						source={captionSource}
						tagsStyles={captionStyles}
					/>
				</View>
			</View>
		</View>
	);
};

export { LightboxCaption };
