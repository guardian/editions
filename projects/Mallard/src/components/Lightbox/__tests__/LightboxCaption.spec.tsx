import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { LightboxCaption } from '../LightboxCaption'

describe('LightboxCaption', () => {
    it('should show a LightboxCaption with a pillar colour', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <LightboxCaption
                caption="Claude Gnapka scores Luton's third and winner"
                pillarColor="#0000FF"
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
