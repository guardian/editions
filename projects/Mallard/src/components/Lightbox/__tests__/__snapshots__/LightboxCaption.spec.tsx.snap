// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightboxCaption should show a LightboxCaption with a pillar colour 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#121212",
      "bottom": 0,
      "opacity": 0.8,
      "position": "absolute",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingTop": 5,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={9}
      bbWidth={11}
      fill="none"
      focusable={false}
      height={9}
      meetOrSlice={0}
      minX={0}
      minY={0}
      rotation={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "marginTop": 4,
          },
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 9,
            "width": 11,
          },
        ]
      }
      vbHeight={9}
      vbWidth={11}
      width={11}
    >
      <RNSVGGroup
        fill={null}
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M5.5 0L10.5 9H0.5L5.5 0Z"
            fill={
              Array [
                0,
                4278190335,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Object {
          "color": "#dcdcdc",
          "paddingBottom": 50,
          "paddingLeft": 2,
        }
      }
    >
      Claude Gnapka scores Luton's third and winner
    </Text>
  </View>
</View>
`;
